ui/pager.go

```diff
@@ -199,6 +202,92 @@ func (m pagerModel) Update(msg tea.Msg) (pagerModel, tea.Cmd) {
 			}
 		default:
 			switch msg.String() {
+			case "e":
+				// First, determine if we can find an editor
+				editor_path, err := getEditor()
+
+				if err != nil {
+					cmd := m.showStatusMessage("No editor found!")
+					cmds = append(cmds, cmd)
+					break
+				}
+
+				// Determine if the current file is stashed or not
+				isStashed := m.currentDocument.markdownType == stashedMarkdown ||
+					m.currentDocument.markdownType == convertedMarkdown
+				
+				// Copy the contents of the document to a temporary file
+				tempFile, err := ioutil.TempFile("", "glow-edit")
```

Comment:
Most of this I/O type stuff should be done in a command so it doesn't block. The command will return a message which you can then handle in this `update` function. Commands and Messages are are a Bubble Tea thing, see [the command tutorial](https://github.com/charmbracelet/bubbletea/tree/master/tutorials/commands).

Reply:
Thanks! I'll definitely check that out over the weekend.

Reply:
> Most of this I/O type stuff should be done in a command so it doesn't block. The command will return a message which you can then handle in this `update` function. Commands and Messages are are a Bubble Tea thing, see [the command tutorial](https://github.com/charmbracelet/bubbletea/tree/master/tutorials/commands).

I know this is closed, but this seemed the relevant place to put this. I just got done rewriting this to use tea.Cmd and tea.Msg... and I didn't even think about how making this non-blocking would be bad. Now, as soon as you open the editor, it jumps right back to trying to view the pager. And there's a fight between the editor and glow painting the screen lol.

tl;dr: I'm 98% sure we want this to be blocking.